src/vorta/views/archive_tab.py

```diff
@@ -831,6 +836,45 @@ def cell_double_clicked(self, row, column):
             if mount_point is not None:
                 QDesktopServices.openUrl(QtCore.QUrl(f'file:///{mount_point}'))
 
+        if column == 4:
+            item = self.archiveTable.item(row, column)
+            self.renamed_archive_original_name = item.text()
+            item.setFlags(item.flags() | QtCore.Qt.ItemFlag.ItemIsEditable)
+            self.archiveTable.editItem(item)
+
+    def cell_changed(self, row, column):
+        # return if this is not a name change
+        if column != 4:
+            return
+
+        item = self.archiveTable.item(row, column)
+        new_name = item.text()
+        profile = self.profile()
+
+        # if the name hasn't changed or if this slot is called when first repopulating the table, do nothing.
+        if new_name == self.renamed_archive_original_name or not self.renamed_archive_original_name:
+            return
+
+        if not new_name:
+            item.setText(self.renamed_archive_original_name)
+            self._set_status(self.tr('Archive name cannot be blank.'))
+            return
+
+        new_name_exists = ArchiveModel.get_or_none(name=new_name, repo=profile.repo)
+        if new_name_exists is not None:
+            self._set_status(self.tr('An archive with this name already exists.'))
```

Comment:
Should `item.setText(self.renamed_archive_original_name)` get called before this line? When I'm testing this, I find the status message is easily missed, and the name stays as the invalid name until an archive refresh.

Reply:
Yes, this is still an issue.